### PR TITLE
Add delete action for templates

### PR DESCRIPTION
**Pull request by andyvesel**
Added `delete` action for Openstack images to API
https://bugzilla.redhat.com/show_bug.cgi?id=1487114
https://github.com/ManageIQ/manageiq-providers-openstack/pull/236

### CODE REVIEW

File: config/api.yml [hunk 1]
```diff
@@ -497,10 +497,18 @@
       :get:
       - :name: read
         :identifier: image_show_list
+      - :name: delete
```
Comment (abellotti) [inline config/api.yml]: probably meant that to be under a :post: section

File: spec/requests/cloud_templates_spec.rb [hunk 2]
```diff
@@ -55,4 +55,24 @@
...
+
+    it "will not delete image unless authorized" do
+      api_basic_authorize
+      ems = FactoryGirl.create(:ems_cloud)
+      image = FactoryGirl.create(:template_cloud, :ext_management_system => ems)
+
+      delete(api_provider_cloud_template_url(nil, ems, image))
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
```
Comment (abellotti) [inline spec/requests/cloud_templates_spec.rb]: will need tests for the POST based deletes, single and bulk.

Comment (andyvesel) [reply spec/requests/cloud_templates_spec.rb]: Thanks for notice, Tests added

File: config/api.yml (hunk 1 above)
Comment (andyvesel) [reply config/api.yml]: Fixed

**Comment by abellotti** [comment]
Thanks @andyvesel for the updates, can you take care of the Rubocop warnings, Thanks.

/cc @jntullo

**Comment by andyvesel** [comment]
@abellotti Rubocop has been fixed

**Comment by andyvesel** [comment]
@abellotti could you review?

**Comment by abellotti** [comment]
Thanks @andyvesel for the API enhancement, LGTM !! 👍

**Pull request merged**